### PR TITLE
VS Code Dockerfile and .json config file 

### PR DESCRIPTION
**Pull request by jeydobson**
Add Dockerfile, .json config and instructions in README

### CODE REVIEW

**Comment by jeydobson** [comment]
Commit message had input from my toddler...

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+FROM mcr.microsoft.com/vscode/devcontainers/base:dev-${VARIANT}
+
+# Install needed packages. Use a separate RUN statement to add your own dependencies.
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install build-essential cmake cppcheck valgrind clang lldb llvm gdb \
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+# [Optional] Uncomment this section to install additional OS packages.
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends clang-format
```
Comment (dpshelio) [inline .devcontainer/Dockerfile]: maybe we want to add here also the cleaning steps of the `apt` to keep the image smaller?
```suggestion
    && apt-get -y install --no-install-recommends clang-format \
    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+To get setup follow the steps here https://code.visualstudio.com/docs/remote/containers-tutorial to:
+   * Install VS Code on your host machine (Mac, Windows, Linux variant)
+   * Install Docker Desktop on your host machine and make sure it is running
+   * Open VS Code and add the `Remote - Containers` extension `ms-vscode-remote.remote-containers`
+   * Click on the green remote containers status button at the bottom left of the window
+   * Then to check everything works select the `Remote-Containers: Try a Sample...` and the C++ example
+   * Go through the README.md and suggested things to try to familiarise yourself  
+ 
+To use the Ubuntu 20.04 dev container you should clone this repository to your host machine and then copy the `.devcontainer` folder including the devcontainer.json and Dockerfile files inside it to the cmake project you want to develop:
+`cp -r .devcontainer /path/to/YourCMakeProject/.`
```
Comment (dpshelio) [inline README.md]:
```suggestion
`cp -r .devcontainer /path/to/YourCMakeProject/`.
```

**Review by dpshelio — APPROVED** [review]
Great, all the steps clear. Only two small things

**Pull request merged**